tests/styleguide.spec.ts

```diff
@@ -1,151 +1,178 @@
+test.describe.configure({ mode: 'serial' });
+
+let page: Page;
+
+async function isolateAndGetSection(sectionId: string) {
+  const isolator = page.getByPlaceholder('Isolate');
+  await expect(isolator).toHaveCount(1);
+  await isolator.fill(sectionId);
+  // sanity check
+  await expect(isolator).toHaveValue(sectionId);
+
+  const section = page.locator(`#sample-section-${sectionId}`);
+  await expect(section).toHaveCount(1);
+  const sectionBox = await section.boundingBox();
+  expect(sectionBox?.height).toBeGreaterThan(0);
+  return section;
+}
+
+test.beforeAll(async ({ browser }) => {
+  page = await browser.newPage();
+  await page.goto('/ide/styleguide?testmode=true');
+  await expect(
+    page.locator('.iris-grid .iris-grid-loading-status')
+  ).toHaveCount(0);
+});
+
+test.afterAll(async () => {
+  await page.close();
+});
 
-  const sampleSections = await page.locator(
+test('Styleguide section count', async () => {
+  const sampleSections = page.locator(
     '.sample-section:not(.sample-section-e2e-ignore)'
   );
-
-  await expect(sampleSections).toHaveCount(sampleSectionIds.length);
+  await expect(sampleSections).toHaveCount(
+    sampleSectionIds.length + isolatedSampleSectionIds.length
+  );
 });
 
-test('UI regression test - Styleguide button section count', async ({
-  page,
-}) => {
-  await page.goto('/ide/styleguide');
-
+test('Styleguide button section count', async () => {
   const buttonSections = await page.locator('[id^="sample-section-buttons-"]');
-
   await expect(buttonSections).toHaveCount(buttonSectionIds.length);
 });
 
-// Iterate over all sample sections and take a screenshot of each one.
-sampleSectionIds.forEach(id => {
-  test(`UI regression test - Styleguide section - ${id}`, async ({ page }) => {
-    // Isolate the section
-    await page.goto(`/ide/styleguide?isolateSection=true#${id}`);
-
-    const sampleSection = page.locator(`#${id}`);
-
-    await expect(sampleSection).toHaveScreenshot(
-      `${id.replace(/^sample-section-/, '')}.png`
-    );
+test.describe('Styleguide section', () => {
+  // Iterate over all sample sections and take a screenshot of each one.
+  isolatedSampleSectionIds.forEach(id => {
+    test(id, async ({ page: page2 }) => {
+      await page2.goto(`/ide/styleguide?testMode=true#${id}`);
+      const sampleSection = page2.locator(`#sample-section-${id}`);
+      await expect(sampleSection).toHaveCount(1);
+      await expect(sampleSection).toHaveScreenshot(`${id}.png`);
+    });
+  });
+  sampleSectionIds.forEach(id => {
+    test(id, async () => {
+      const sampleSection = await isolateAndGetSection(id);
+      await expect(sampleSection).toHaveScreenshot(`${id}.png`);
+    });
   });
 });
 
-buttonSectionIds.forEach((id, i) => {
-  test(`Buttons regression test - ${id}`, async ({ page }) => {
-    // Isolate the section
-    await page.goto(`/ide/styleguide?isolateSection=true#${id}`);
+test.describe('Buttons regression test', () => {
```

Comment:
Does a failing test in the forEach cause remaining tests to be skipped, or will all still run?

Reply:
It does cause it to be skipped over since it is in serial and Playwright doesn't have a feature to continue after a failed test in serial mode (even with soft assertions). However, I've made it so that the test will only be serial for CI. When running locally, it will be in parallel and continue with all the tests.